app/src/main/java/org/wikipedia/main/MainActivity.kt

```diff
@@ -19,10 +20,19 @@ import org.wikipedia.util.FeedbackUtil
 import org.wikipedia.util.ResourceUtil
 
 class MainActivity : SingleFragmentActivity<MainFragment>(), MainFragment.Callback {
+    interface Callback {
+        fun onActivityListsImportResult(uri: Uri)
+    }
+
     private lateinit var binding: ActivityMainBinding
 
     private var controlNavTabInFragment = false
     private val onboardingLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { }
+    var importCallback: Callback? = null
+    val filePickerLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+        val uri: Uri = it.data?.data!!
+        importCallback?.onActivityListsImportResult(uri)
+    }
```

Comment:
Would it be possible to move the `launcher` to `ReadingListsFragment`? Not sure what's the benefit of putting the launcher in the `MainActivity`.

Reply:
Good suggestion @cooltey and done.